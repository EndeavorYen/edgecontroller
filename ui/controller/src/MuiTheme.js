// Copyright 2019 Intel Corporation and Smart-Edge.com, Inc. All rights reserved
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import {createMuiTheme} from "@material-ui/core";
import {blue, indigo} from "@material-ui/core/colors";

const theme = createMuiTheme({
  palette: {
    secondary: {
      main: blue[900]
    },
    primary: {
      main: indigo[700]
    },
    grey: {
      '100': '#F8F8F8',
    }
  },
  typography: {
    useNextVariants: true,
    // Use the system font instead of the default Roboto font.
    fontFamily: [
      '"Lato"',
      'sans-serif'
    ].join(',')
  }
});

export default theme
