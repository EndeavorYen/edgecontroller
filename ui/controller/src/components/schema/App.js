export default {
  schema: {
    type: "object",
    title: "App",
    required: [
      "type",
      "name",
      "version",
      "vendor",
      "cores",
      "memory",
      "source"
    ],
    properties: {
      id: {
        type: "string",
        title: "ID",
        readonly: true
      },
      type: {
        type: "string",
        title: "Type",
        enum: [
          "container",
          "vm"
        ]
      },
      name: {
        type: "string",
        title: "Name"
      },
      version: {
        type: "string",
        title: "Version"
      },
      vendor: {
        type: "string",
        title: "Vendor"
      },
      description: {
        type: "string",
        title: "Description"
      },
      cores: {
        type: "number",
        title: "Cores",
      },
      memory: {
        type: "number",
        title: "Memory",
      },
      ports: {
        type: "array",
        title: "Ports",
        items: {
          type: "object",
          title: "Port",
          properties: {
            port: {
              type: "number",
              title: "Port",
            },
            protocol: {
              type: "string",
              title: "Protocol",
            },
          },
        },
      },
      source: {
        type: "string",
        title: "Source",
      }
    }
  },
  form: [
    "*"
  ]
};
