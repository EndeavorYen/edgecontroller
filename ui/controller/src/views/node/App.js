import React, { Component } from 'react';
import ApiClient from "../../api/ApiClient";
import { withSnackbar } from 'notistack';

class AppView extends Component {
  constructor(props) {
    super(props);

    this.state = {
      loaded: false,
      error: null,
      app: {},
      policy: {},
    };
  }

  // GET /nodes/:node_id/apps/:app_id
  getNodeApp = () => {
    const { nodeID, appID } = this.props;

    ApiClient.get(`/nodes/${nodeID}/apps/${appID}`)
      .then((resp) => {
        this.setState({
          loaded: true,
          app: resp.data.apps || [],
        })
      })
      .catch((err) => {
        this.setState({
          loaded: true,
        });

        this.props.enqueueSnackbar(`${err.toString()}. Please try again later.`, { variant: 'error' });
      });
  }

  // PATCH /nodes/:node_id/apps/:app_id
  updateNodeApp = () => {
    const { nodeID, appID } = this.props;
    const { app } = this.state;

    ApiClient.patch(`/nodes/${nodeID}/apps/${appID}`, app)
      .then((resp) => {
        this.setState({
          loaded: true,
        })
      })
      .catch((err) => {
        this.setState({
          loaded: true,
        });

        this.props.enqueueSnackbar(`${err.toString()}. Please try again later.`, { variant: 'error' });
      });
  }

  // GET /nodes/:node_id/apps/:app_id/policy
  getNodeAppPolicy = () => {
    const { nodeID, appID } = this.props;

    ApiClient.get(`/nodes/${nodeID}/apps/${appID}/policy`)
      .then((resp) => {
        this.setState({
          loaded: true,
          policy: resp.data || {},
        })
      })
      .catch((err) => {
        this.setState({
          loaded: true,
        });

        this.props.enqueueSnackbar(`${err.toString()}. Please try again later.`, { variant: 'error' });
      });
  }

  // PATCH /nodes/:node_id/apps/:app_id/policy
  updateNodeAppPolicy = () => {
    const { nodeID, appID } = this.props;
    const { policy } = this.state;

    ApiClient.patch(`/nodes/${nodeID}/apps/${appID}/policy`, policy)
      .then((resp) => {
        this.setState({
          loaded: true,
        })
      })
      .catch((err) => {
        this.setState({
          loaded: true,
        });

        this.props.enqueueSnackbar(`${err.toString()}. Please try again later.`, { variant: 'error' });
      });
  }

  // DELETE /nodes/:node_id/apps/:app_id/policy
  deleteNodeAppPolicy = () => {
    const { nodeID, appID } = this.props;

    ApiClient.delete(`/nodes/${nodeID}/apps/${appID}/policy`)
      .then((resp) => {
        this.setState({
          loaded: true,
        })
      })
      .catch((err) => {
        this.setState({
          loaded: true,
        });

        this.props.enqueueSnackbar(`${err.toString()}. Please try again later.`, { variant: 'error' });
      });
  }

  componentDidMount() {
    this.getNodeApp();
  }

  render() {
    const {
      loaded,
      app,
    } = this.state;

    if (!loaded) {
      return <React.Fragment>Loading ...</React.Fragment>
    }

    return (
      <React.Fragment>
        {JSON.stringify(app)}
      </React.Fragment>
    );
  }
};

export default withSnackbar(AppView);
