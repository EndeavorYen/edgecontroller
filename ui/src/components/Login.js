import React,  { Component } from 'react';
import withStyles from '@material-ui/core/styles/withStyles';
import { withRouter } from 'react-router-dom'
import CssBaseline from '@material-ui/core/CssBaseline';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import Button from '@material-ui/core/Button';
import FormControl from '@material-ui/core/FormControl';
import Avatar from '@material-ui/core/Avatar';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Checkbox from '@material-ui/core/Checkbox';
import Input from '@material-ui/core/Input';
import InputLabel from '@material-ui/core/InputLabel';
import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
import Auth from './Auth';

const backgroundShape = require('../images/shape.svg');
const logo = require('../images/logo.svg');
const numeral = require('numeral');

numeral.defaultFormat('0');
const styles = theme => ({
  root: {
    flexGrow: 1,
    backgroundColor: theme.palette.secondary['A100'],
    overflow: 'hidden',
    background: `url(${backgroundShape}) no-repeat`,
    backgroundSize: 'cover',
    backgroundPosition: '0 400px',
    marginTop: 10,
    padding: 20,
    paddingBottom: 500
  },
  grid: {
    margin: `0 ${theme.spacing.unit * 2}px`
  },
  smallContainer: {
    width: '60%'
  },
  bigContainer: {
    width: '80%'
  },
  logo: {
    marginBottom: 24,
    display: 'flex',
    justifyContent: 'center'
  },
  stepContainer: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center'
  },
  stepGrid: {
    width: '80%'
  },
  buttonBar: {
    marginTop: 32,
    display: 'flex',
    justifyContent: 'center'
  },
  button: {
    backgroundColor: theme.palette.primary['A100']
  },
  backButton: {
    marginRight: theme.spacing.unit,
  },
  outlinedButtom: {
    textTransform: 'uppercase',
    margin: theme.spacing.unit
  },
  stepper: {
    backgroundColor: 'transparent'
  },
  paper: {
    padding: theme.spacing.unit * 3,
    textAlign: 'left',
    color: theme.palette.text.secondary
  },
  topInfo: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginBottom: 42
  },
  formControl: {
    width: '100%'
  },
  selectEmpty: {
    marginTop: theme.spacing.unit * 2,
  }
});

class LoginForm extends Component {
  handleInputChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  handleLogin = event => {
    event.preventDefault();

    const {email, password} = this.state;

    if (!Auth.login(email, password)) {
      alert('Auth Failed');
    }

    setTimeout(() => this.props.history.push('/home'), 500)
  };

  constructor(props) {
    super(props);

    // This binding is necessary to make `this` work in the callback
    this.handleInputChange = this.handleInputChange.bind(this);
    this.handleLogin = this.handleLogin.bind(this);
  }

  componentDidMount() {

  };

  render() {

    const { classes } = this.props;
    return (
      <React.Fragment>
        <CssBaseline />
        <div className={classes.root}>
          <Grid container justify="center">
            <Grid spacing={24} alignItems="center" justify="center" container className={classes.grid}>
              <Grid item xs={12}>
                <div className={classes.logo}>
                  <img width={100} height={100} src={logo} alt="" />
                </div>
                <div className={classes.stepContainer}>
                  <div className={classes.smallContainer}>
                    <main className={classes.main}>
                      <CssBaseline />
                      <Paper className={classes.paper}>
                        <Avatar className={classes.avatar}>
                          <LockOutlinedIcon />
                        </Avatar>
                        <Typography component="h1" variant="h5">
                          Controller Login
                        </Typography>
                        <form onSubmit={this.handleLogin} className={classes.form}>
                          <FormControl margin="normal" required fullWidth>
                            <InputLabel htmlFor="email">Email Address</InputLabel>
                            <Input id="email" name="email" autoComplete="email" onChange={this.handleInputChange} autoFocus />
                          </FormControl>
                          <FormControl margin="normal" required fullWidth>
                            <InputLabel htmlFor="password">Password</InputLabel>
                            <Input name="password" type="password" id="password" onChange={this.handleInputChange} autoComplete="current-password" />
                          </FormControl>
                          <FormControlLabel
                            control={<Checkbox value="remember" color="primary" />}
                            label="Remember me"
                          />
                          <Button
                            type="submit"
                            fullWidth
                            variant="contained"
                            color="primary"
                            className={classes.submit}
                          >
                            Sign in
                          </Button>
                        </form>
                      </Paper>
                    </main>
                  </div>
                </div>
              </Grid>
            </Grid>
          </Grid>
        </div>
      </React.Fragment>
    )
  }
}

export default withRouter(withStyles(styles)(LoginForm))
