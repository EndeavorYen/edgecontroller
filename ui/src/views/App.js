import React, { Component } from 'react';
import ApiClient from '../api/ApiClient';
import { SchemaForm, utils } from 'react-schema-form';
import AppSchema from '../components/schema/App';
import Topbar from '../components/Topbar';
import withStyles from '@material-ui/core/styles/withStyles';
import { withSnackbar } from 'notistack';
import {
  Grid,
  Button
} from '@material-ui/core';

const styles = theme => ({
  root: {
    flexGrow: 1,
    backgroundColor: theme.palette.grey['A500'],
    overflow: 'hidden',
    backgroundSize: 'cover',
    backgroundPosition: '0 400px',
    marginTop: 20,
    padding: 20,
    paddingBottom: 200
  },
  grid: {
    paddingLeft: '20%',
    paddingRight: '20%'
  },
  gridSaveButton: {
    textAlign: 'right',
  }
});

class AppView extends Component {
  constructor(props) {
    super(props);

    this.state = {
      loaded: false,
      error: null,
      showErrors: true,
      app: {},
    };
  }

  // GET /apps/:app_id
  getApp = () => {
    const { match } = this.props;

    const appID = match.params.id;

    ApiClient.get(`/apps/${appID}`)
      .then((resp) => {
        this.setState({
          loaded: true,
          app: resp.data || {},
        });
      })
      .catch((err) => {
        this.setState({
          loaded: true,
        });

        this.props.enqueueSnackbar(`${err.toString()}. Please try again later.`, { variant: 'error' });
      });
  }

  // PATCH /apps/:app_id
  updateApp = () => {
    const { match } = this.props;
    const { app } = this.state;

    const appID = match.params.id;

    ApiClient.patch(`/apps/${appID}`, app)
      .then((resp) => {
        this.setState({
          loaded: true,
        });

        this.props.enqueueSnackbar(`Successfully updated application.`, { variant: 'success' });
      })
      .catch((err) => {
        this.setState({
          loaded: true,
        });

        this.props.enqueueSnackbar(`${err.toString()}. Please try again later.`, { variant: 'error' });
      });
  }

  onModelChange = (key, val) => {
    const { app } = this.state;

    const newApp = app;

    utils.selectOrSet(key, newApp, val);

    this.setState({ app: newApp });
  }

  componentDidMount() {
    this.getApp();
  }

  render() {
    const { location: { pathname: currentPath }, classes } = this.props;

    const {
      loaded,
      showErrors,
      app,
    } = this.state;

    if (!loaded) {
      return <React.Fragment>Loading ...</React.Fragment>
    }

    return (
      <React.Fragment>
        <Topbar currentPath={currentPath} />
        <div className={classes.root}>
          <Grid
            container
            justify="center"
            alignItems="flex-end"
            spacing={24}
            className={classes.grid}
          >
            <Grid item xs={12}>
              <SchemaForm
                schema={AppSchema.schema}
                form={AppSchema.form}
                model={app}
                onModelChange={this.onModelChange}
                showErrors={showErrors}
              />
            </Grid>
            <Grid item xs={12} className={classes.gridSaveButton}>
              <Button
                onClick={this.updateApp}
                variant="outlined"
                color="primary"
              >
                Save
            </Button>
            </Grid>
          </Grid>

        </div>
      </React.Fragment>
    );
  }
}

export default withStyles(styles)(withSnackbar(AppView));
