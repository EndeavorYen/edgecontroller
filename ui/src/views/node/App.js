import React, { Component } from 'react';
import ApiClient from "../../api/ApiClient";

export default class AppView extends Component {
  constructor(props) {
    super(props);

    this.state = {
      loaded: false,
      errored: false,
      error: null,
      app: {},
      policy: {},
    };
  }

  // GET /nodes/:node_id/apps/:app_id
  getNodeApp = () => {
    const { nodeID, appID } = this.props;

    ApiClient.get(`/nodes/${nodeID}/apps/${appID}`)
      .then((resp) => {
        this.setState({
          loaded: true,
          app: resp.data || [],
        })
      })
      .catch((err) => {
        this.setState({
          loaded: true,
          errored: true,
          error: err,
        });
      });
  }

  // PATCH /nodes/:node_id/apps/:app_id
  updateNodeApp = () => {
    const { nodeID, appID } = this.props;
    const { app } = this.state;

    ApiClient.patch(`/nodes/${nodeID}/apps/${appID}`, app)
      .then((resp) => {
        this.setState({
          loaded: true,
        })
      })
      .catch((err) => {
        this.setState({
          loaded: true,
          errored: true,
          error: err,
        });
      });
  }

  // GET /nodes/:node_id/apps/:app_id/policy
  getNodeAppPolicy = () => {
    const { nodeID, appID } = this.props;

    ApiClient.get(`/nodes/${nodeID}/apps/${appID}/policy`)
      .then((resp) => {
        this.setState({
          loaded: true,
          policy: resp.data || {},
        })
      })
      .catch((err) => {
        this.setState({
          loaded: true,
          errored: true,
          error: err,
        });
      });
  }

  // PATCH /nodes/:node_id/apps/:app_id/policy
  updateNodeAppPolicy = () => {
    const { nodeID, appID } = this.props;
    const { policy } = this.state;

    ApiClient.patch(`/nodes/${nodeID}/apps/${appID}/policy`, policy)
      .then((resp) => {
        this.setState({
          loaded: true,
        })
      })
      .catch((err) => {
        this.setState({
          loaded: true,
          errored: true,
          error: err,
        });
      });
  }

  // DELETE /nodes/:node_id/apps/:app_id/policy
  deleteNodeAppPolicy = () => {
    const { nodeID, appID } = this.props;

    ApiClient.delete(`/nodes/${nodeID}/apps/${appID}/policy`)
      .then((resp) => {
        this.setState({
          loaded: true,
        })
      })
      .catch((err) => {
        this.setState({
          loaded: true,
          errored: true,
          error: err,
        });
      });
  }

  componentDidMount() {
    this.getNodeApp();
  }

  render() {
    const {
      loaded,
      errored,
      error,
      app,
    } = this.state;

    if (!loaded) {
      return <React.Fragment>Loading ...</React.Fragment>
    }

    if (errored) {
      return <React.Fragment>{error.toString()}</React.Fragment>
    }

    return (
      <React.Fragment>
        {JSON.stringify(app)}
      </React.Fragment>
    );
  }
};
