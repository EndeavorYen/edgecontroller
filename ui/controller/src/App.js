import React, { Component } from 'react';
import { BrowserRouter } from "react-router-dom";
import { MuiThemeProvider } from '@material-ui/core/styles';
import './App.css';
import Routes from './routes'
import MuiTheme from './MuiTheme';
import '@material-ui/icons';
import 'typeface-lato';
import 'typeface-roboto';
import { SnackbarProvider } from "notistack";
import CssBaseline from '@material-ui/core/CssBaseline';
import ErrorBoundary from './components/common/ErrorBoundary';

class App extends Component {
  render() {
    return (
      <React.Fragment>
        <CssBaseline />
        <SnackbarProvider maxSnack={3}
          anchorOrigin={{
            vertical: 'top',
            horizontal: 'right'
          }}
        >
          <div>
            <ErrorBoundary>
              <MuiThemeProvider theme={MuiTheme}>
                <BrowserRouter>
                  <Routes />
                </BrowserRouter>
              </MuiThemeProvider>
            </ErrorBoundary>
          </div>
        </SnackbarProvider>
      </React.Fragment>
    );
  }
}

export default App;
