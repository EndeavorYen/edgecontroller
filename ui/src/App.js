import React, { Component } from 'react';
import { BrowserRouter } from "react-router-dom";
import { MuiThemeProvider } from '@material-ui/core/styles';
import './App.css';
import Routes from './routes'
import MuiTheme from './MuiTheme';
import '@material-ui/icons';
import 'typeface-lato';
import 'typeface-roboto';
import {SnackbarProvider} from "notistack";

class App extends Component {
  render() {
    return (
      <SnackbarProvider maxSnack={3}
        anchorOrigin={{
          vertical: 'top',
          horizontal: 'right'
        }}
      >
        <div>
          <MuiThemeProvider theme={MuiTheme}>
            <BrowserRouter>
              <Routes />
            </BrowserRouter>
          </MuiThemeProvider>
        </div>
      </SnackbarProvider>
    );
  }
}

export default App;
