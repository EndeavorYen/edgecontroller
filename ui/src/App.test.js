import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { mount, shallow } from 'enzyme';
import MuiThemeProvider from '@material-ui/core/styles/MuiThemeProvider';
import Routes from "./routes";
import MuiTheme from "./MuiTheme";
import Login from "./components/Login";

// Use React Dom to render App Component
it('renders without crashing (ReactDOM)', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

// Use Enzyme Shallow rendering for App Component
it('renders without crashing (Shallow)', () => {
  const sut = shallow(<App />);
  expect(sut.contains('div'));
});

// Use Enzyme Mount for App Component
it('renders without crashing (Mount)', () => {
  const sut = mount(<App />);
  const ThemeWrapper = (
    <MuiThemeProvider theme={MuiTheme}>
      <Routes />
    </MuiThemeProvider>
  );

  expect(sut.contains(ThemeWrapper)).toEqual(true);
});

// Validate Route exists

it('Prompts Login screen', () => {
  const sut = mount(<App />);

  expect(sut.contains(Login));

  // Expect total of 3 input field
  expect(sut.find('input')).toHaveLength(3);
});
