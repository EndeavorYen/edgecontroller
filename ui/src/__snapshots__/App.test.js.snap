// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smoke Tests for Main App.js Validate App Screen snapshot 1`] = `
<div>
  <div>
    <div
      className="LoginForm-root-274"
    >
      <div
        className="MuiGrid-container-287 MuiGrid-justify-xs-center-304"
      >
        <div
          className="MuiGrid-container-287 MuiGrid-spacing-xs-24-311 MuiGrid-align-items-xs-center-295 MuiGrid-justify-xs-center-304 LoginForm-grid-275"
        >
          <div
            className="MuiGrid-item-288 MuiGrid-grid-xs-12-327"
          >
            <div
              className="LoginForm-logo-278"
            >
              <img
                alt=""
                height={100}
                src={
                  Object {
                    "ReactComponent": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "__esModule": true,
                    "default": "logo.svg",
                  }
                }
                width={100}
              />
            </div>
            <div
              className="LoginForm-stepContainer-279"
            >
              <div
                className="LoginForm-smallContainer-276"
              >
                <main>
                  <div
                    className="MuiPaper-root-384 MuiPaper-elevation2-388 MuiPaper-rounded-385 LoginForm-paper-283"
                  >
                    <div
                      className="MuiAvatar-root-411 MuiAvatar-colorDefault-412"
                    >
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root-414"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <g
                          fill="none"
                        >
                          <path
                            d="M0 0h24v24H0V0z"
                          />
                          <path
                            d="M0 0h24v24H0V0z"
                            opacity=".87"
                          />
                        </g>
                        <path
                          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
                        />
                      </svg>
                    </div>
                    <h1
                      className="MuiTypography-root-423 MuiTypography-h5-439"
                    >
                      Controller Login
                    </h1>
                    <form
                      autoComplete="off"
                      onSubmit={[Function]}
                    >
                      <div
                        className="MuiFormControl-root-459 MuiFormControl-marginNormal-460 MuiFormControl-fullWidth-462"
                      >
                        <label
                          className="MuiFormLabel-root-474 MuiFormLabel-required-479 MuiInputLabel-required-467 MuiInputLabel-root-463 MuiInputLabel-formControl-468 MuiInputLabel-animated-471"
                          data-shrink={false}
                          htmlFor="username"
                        >
                          Username
                          <span
                            className="MuiFormLabel-asterisk-480"
                          >
                             *
                          </span>
                        </label>
                        <div
                          className="MuiInputBase-root-494 MuiInput-root-481 MuiInput-underline-485 MuiInputBase-formControl-495 MuiInput-formControl-482"
                          onClick={[Function]}
                        >
                          <input
                            aria-invalid={false}
                            autoComplete="off"
                            autoFocus={true}
                            className="MuiInputBase-input-504 MuiInput-input-489"
                            disabled={false}
                            id="username"
                            name="username"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            required={true}
                            type="text"
                          />
                        </div>
                      </div>
                      <div
                        className="MuiFormControl-root-459 MuiFormControl-marginNormal-460 MuiFormControl-fullWidth-462 LoginForm-passwordBox-286"
                      >
                        <label
                          className="MuiFormLabel-root-474 MuiFormLabel-required-479 MuiInputLabel-required-467 MuiInputLabel-root-463 MuiInputLabel-formControl-468 MuiInputLabel-animated-471"
                          data-shrink={false}
                          htmlFor="password"
                        >
                          Password
                          <span
                            className="MuiFormLabel-asterisk-480"
                          >
                             *
                          </span>
                        </label>
                        <div
                          className="MuiInputBase-root-494 MuiInput-root-481 MuiInput-underline-485 MuiInputBase-formControl-495 MuiInput-formControl-482"
                          onClick={[Function]}
                        >
                          <input
                            aria-describedby="component-error-text"
                            aria-invalid={false}
                            autoComplete="off"
                            className="MuiInputBase-input-504 MuiInput-input-489 MuiInputBase-inputType-507 MuiInput-inputType-492"
                            disabled={false}
                            id="password"
                            name="password"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            required={true}
                            type="password"
                          />
                        </div>
                      </div>
                      <button
                        className="MuiButtonBase-root-537 MuiButton-root-511 MuiButton-contained-522 MuiButton-containedPrimary-523 MuiButton-raised-525 MuiButton-raisedPrimary-526 MuiButton-fullWidth-536 LoginForm-buttonBar-281"
                        disabled={false}
                        onBlur={[Function]}
                        onContextMenu={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        tabIndex="0"
                        type="submit"
                      >
                        <span
                          className="MuiButton-label-512"
                        >
                          Sign in
                        </span>
                        <span
                          className="MuiTouchRipple-root-540"
                        />
                      </button>
                    </form>
                  </div>
                </main>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
